controlplane/kubeadm/controllers/controller.go

```diff
@@ -324,7 +327,7 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 	}
 
 	// If we've made it this far, we can assume that all ownedMachines are up to date
-	numMachines := len(ownedMachines)
+	numMachines := len(controlPlane.Machines)
```

Comment:
Make this a method on `controlPlane` struct?

Reply:
Reverted to use ownedMachines.